tests/stub/shared.py

```diff
@@ -382,3 +383,20 @@ def pick_address(adapter_):
             ips.append(address)
 
     return ips
+
+
+def dns_resolve(addr):
+    return socket.gethostbyname_ex(addr)[2]
```

Comment:
Maybe:
```
_, _, ips = socket.gethostbyname_ex(addr)
return ips
```

Or something similar. It makes easier to realize what's the method is returning.